cmd/migration-managerd/internal/api/api_source.go

```diff
@@ -192,13 +194,23 @@ func sourcesPost(d *Daemon, r *http.Request) response.Response {
 		return response.SmartError(fmt.Errorf("Failed creating source %q: %w", source.Name, err))
 	}
 
-	// Trigger a scan of this new source for instances.
-	err = d.syncOneSource(d.ShutdownCtx, src)
+	d.checkSourceConnectivity()
+
+	// Get the source's connectivity status to return to the client.
+	currentSource, err := d.source.GetByName(r.Context(), source.Name)
```

Comment:
Why is it necessary to get the source again, if `Create` already returns the just created source?

Reply:
It was because while the connectivity status was updated in the database, the object we had would then be stale. This bit of logic has since been refactored away.